libfdt/meson.build

```diff
@@ -26,27 +26,33 @@ else
 endif
 
 link_args += version_script
-libfdt = both_libraries(
+libfdt = library(
```

Comment:
I don't love the idea of avoiding `both_libraries` which seems to have some builtin logic to make our life easier.  I thought `both_libraries` already had support for selecting just one or the other libraro to build.

Reply:
That fixes the static builds of `dtc` on NixOS.
The patch also works for dynamic builds.

I'm not sure I understand how `both_libraries` is supposed to work though.

Reply:
So the main difference between `library()` and `both_libraries()` is that `both_libraries()` _always_ builds both the static and dynamic versions of a library. Ultimately that is what I run into problems with, meson-python tries to map all the build artifacts into a Python package, but a static archive doesn't map. I'm guessing NixOS has a similar issue that it doesn't like the static archive.

One reason to use `both_libraries()` is it makes it simpler to force Meson to statically link your binaries. Since it always builds a static archive, `both_libraries()` returns the `both_libs` type which lets you do `lib.get_static_lib()` to get a direct handle. That is used later in this file to change which library is linked based on the custom `static-build` option. With just `library()` you can't access the type, and Meson decides whether to use the static or shared library.

Meson does provide some core options that let the builder influence the library type. `--default-library {shared,static,both}` and `--prefer-static`, but neither of them let you enforce a static build. I'm not sure what the value is of an option to _require_ static build, but it does complicate things.

`library()` function: https://mesonbuild.com/Reference-manual_functions.html#library
`both_library()` function: https://mesonbuild.com/Reference-manual_functions.html#both_libraries
`--default-library/--prefer-static`: https://mesonbuild.com/Builtin-options.html#core-options